### ISSUE TITLE
Getting the following puppet deprecation when reusing your code

### ISSUE THREAD

**Issue by nick-george** [voxpupuli/puppet-vault_lookup#22]
Hi there, 

I've been using your code to pull not-so-secret bits of data out of the vault to use as facts. When I run facter, I get the following deprecation warning out of Puppet: 

`Warning: Puppet::SSL::Host is deprecated and will be removed in a future release of Puppet.
   (location: /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/ssl/host.rb:235:in `initialize')
`
I'm getting this warning on puppet agent version 6.11.1. I expect your code will also be affected once  V7 of puppet is released. 

Cheers, 
Nick

**Comment by ekohl** [comment]
I've seen the same thing but I'm not using this module. Are you by any chance also using PuppetDB?

**Comment by nick-george** [comment]
Hi @ekohl,

It's definitely being generated from the following line of code: 

`connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)`

Here's the PuppetLabs ticket where they talk about deprecating it. 
https://tickets.puppetlabs.com/browse/PUP-9513?attachmentViewMode=list

Cheers, 
Nick

**Comment by ekohl** [comment]
Here's how I used a non-deprecated function: https://github.com/theforeman/puppet-foreman/blob/f2733160ceeced507faa1bbde1118a812f0e1216/lib/puppet/functions/foreman/enc.rb#L44-L46

Could you come up with a (tested) patch?

**Comment by alexjfisher** [comment]
@ekohl Have you seen https://github.com/voxpupuli/puppet-vault_lookup/pull/18 ?

**Comment by nick-george** [comment]
Oh nice, apologies for not noticing this PR before raising the ticket! Shall I close it? 

I've also noticed that I have had to use the code in https://github.com/voxpupuli/puppet-vault_lookup/pull/21 in order to make proper use of this module. Fingers crossed it gets merged soon!

I've also found that puppet is extremely sensitive about how the secrets from Vault are used. It seems that if there's any hint of needing access to the content of the secret during catalog compilation (e.g basing a conditional on the value of a secret), Puppet gets upset. This is probably more of an issue for Puppetlabs to properly document, but it could be worth putting into the docs of this module. It is for this reason that I've had to plonk a lot of my (not so secret) secrets into facts, while 'hiding' the real secrets from facter. Then I pull them out for real using this module during the catalog run. It's a bit painful, but it seems to work. 

Cheers, 
Nick
